### PR TITLE
Vue datetime component

### PR DESCRIPTION
**Pull request by skjnldsv**
- [x] Use locale to properly format the date
- [x] Datepicker
- [x] VcardTime - Moment - Datepicker data sharing
- [x] Datepicker design https://github.com/nextcloud/vue-components/pull/23

__
- [ ] ~~Open the datepicker on click of the formatted date @sleepypioneer~~
- [x] Moment format date without year

### CODE REVIEW

File: src/components/Properties/PropertyDateTime.vue [hunk 1]
```diff
@@ -41,34 +41,79 @@
...
+/**
+ * Override format function and use this since this
+ * inside a function declaration will represent the
+ * location of the call. So this = DatetimePicker.
+ * Therefore we can use any props we pass through datetime-picker
+ */
+DatetimePicker.methods.stringify = function() {
+	return formatDateTime(this.$parent.localValue, this.type, this.$parent.locale)
+}
+
```
Comment (sleepypioneer) [inline src/components/Properties/PropertyDateTime.vue]: Nice so you add this into the date picker as a method!

File: src/components/Properties/PropertyDateTime.vue [hunk 2]
```diff
@@ -195,6 +201,8 @@ export default {
 		// default load e.g. fr-fr
 		import('moment/locale/' + this.locale)
```
Comment (sleepypioneer) [inline src/components/Properties/PropertyDateTime.vue]: Interested to know more about how this works

File: src/components/Properties/PropertyDateTime.vue [hunk 3]
```diff
@@ -195,6 +201,8 @@ export default {
...
 			.then(e => {
+				// force locale change to update
+				// the component once done loading
 				this.locale = locale
 			})
 			.catch(e => {
```
Comment (sleepypioneer) [inline src/components/Properties/PropertyDateTime.vue]: So does update-input emit for this to happen?

File: src/components/Properties/PropertyDateTime.vue (hunk 1 above)
Comment (skjnldsv) [reply src/components/Properties/PropertyDateTime.vue]: I replace an existing method! :)

File: src/components/Properties/PropertyDateTime.vue (hunk 3 above)
Comment (skjnldsv) [reply src/components/Properties/PropertyDateTime.vue]: I don't get your question :)

**Pull request merged**